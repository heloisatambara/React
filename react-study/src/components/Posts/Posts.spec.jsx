import { render, screen } from '@testing-library/react';
import { Posts } from '.';
// another way to import: const {render} = require('@testing-library/react';)

const props = {
    posts: [
        {
            id:1,
            title:"t1",
            body:"b1",
            cover:"img/img1.png"
        },
        {
            id:2,
            title:"t2",
            body:"b2",
            cover:"img/img2.png"
        },
        {
            id:3,
            title:"t3",
            body:"b3",
            cover:"img/img3.png"
        },
    ]
}

describe('<Posts/>', () => {
    it('should render posts', () => {
        render(<Posts {...props}/>);

        expect(screen.getAllByRole('heading', {name: /t/i}))
            .toHaveLength(3);
        expect(screen.getAllByRole('img', {name: /t/i}))
            .toHaveLength(3);
        expect(screen.getAllByText(/b/i))
            .toHaveLength(3);
        expect(screen.getByRole('img', {name: /t3/i}))
            .toHaveAttribute('src', 'img/img3.png');
    });

    it('should not render posts', () => {
        render(<Posts/>);

        expect(screen.queryByRole('heading', {name: /t/i}))
            .not.toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const {container} = render(<Posts {...props}/>);

        expect(container).toMatchSnapshot();
    });
}); 