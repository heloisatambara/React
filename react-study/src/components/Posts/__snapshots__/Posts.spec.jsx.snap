// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="title 1"
        src="img/img1.png"
      />
      <div
        class="post-content"
      >
        <h2>
          title 1
        </h2>
        <p>
          body 1
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="t2"
        src="img/img2.png"
      />
      <div
        class="post-content"
      >
        <h2>
          t2
        </h2>
        <p>
          b2
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="t3"
        src="img/img3.png"
      />
      <div
        class="post-content"
      >
        <h2>
          t3
        </h2>
        <p>
          b3
        </p>
      </div>
    </div>
  </div>
</div>
`;
